### PR TITLE
Rename proto.IsTransactional to proto.IsTransactionWrite.

### PR DESCRIPTION
**Pull request by petermattis**
Update comment to make purpose of proto.IsTransactionWrite clearer.

### CODE REVIEW

File: proto/api.go [hunk 1]
```diff
@@ -110,9 +110,9 @@ func IsReadOnly(args Request) bool {
 	return IsRead(args) && !IsWrite(args)
 }
 
-// IsTransactional returns true if the request can be part of a
-// transaction.
-func IsTransactional(args Request) bool {
+// IsTransactionWrite returns true if the request produces write
+// intents when used within a transaction.
+func IsTransactionWrite(args Request) bool {
 	return (args.flags() & isTxn) != 0
```
Comment (bdarnell) [inline proto/api.go]: Also rename the isTxn flag.

**Comment by bdarnell** [comment]
LGTM

File: proto/api.go (hunk 1 above)
Comment (petermattis) [inline proto/api.go]: Good point. Done.

**Comment by tschottdorf** [comment]
LGTM

**Comment by spencerkimball** [comment]
LGTM

**Pull request merged**